rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -762,8 +762,18 @@ object ProcNormalizeMatcher {
           )
       }
 
-      case p: PIf     => normalizeIfElse(p.proc_1, p.proc_2, new PNil())
-      case p: PIfElse => normalizeIfElse(p.proc_1, p.proc_2, p.proc_3)
+      case p: PIf     =>
+        normalizeIfElse(p.proc_1, p.proc_2, new PNil(), input.copy(par = VectorPar()))
+          .map(normalized => ProcVisitOutputs(
+            input.par ++ normalized.par,
+            normalized.knownFree
+          ))
+      case p: PIfElse =>
+        normalizeIfElse(p.proc_1, p.proc_2, p.proc_3, input.copy(par = VectorPar()))
+          .map(normalized => ProcVisitOutputs(
+            input.par ++ normalized.par,
+            normalized.knownFree
+          ))
```

Comment:
You should be able to reduce this code duplication

Reply:
Done.